Tests/UnitTests/Core/Utilities/GridIterationTests.cpp

```diff
@@ -0,0 +1,373 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Utilities/Grid.hpp"
+#include "Acts/Utilities/GridIterator.hpp"
+
+#include <array>
+
+namespace Acts::Test {
+
+BOOST_AUTO_TEST_CASE(grid_iteration_test_1d_global_operators) {
+  const std::size_t nBins = 10ul;
+  Acts::detail::EquidistantAxis xAxis(0, 100, nBins);
+  Acts::Grid<double, Acts::detail::EquidistantAxis> grid(
+      std::make_tuple(std::move(xAxis)));
+
+  BOOST_CHECK_EQUAL(grid.size(true), nBins + 2ul);
+  
+  Acts::GridGlobalIterator<double, Acts::detail::EquidistantAxis> gridStart =
+      grid.begin();
+  Acts::GridGlobalIterator<double, Acts::detail::EquidistantAxis> gridStop =
+      grid.end();
+
+  BOOST_CHECK_EQUAL(gridStart == gridStop, false);
+  BOOST_CHECK_EQUAL(gridStart != gridStop, true);
+  BOOST_CHECK_EQUAL(gridStart < gridStop, true);
+  BOOST_CHECK_EQUAL(gridStart <= gridStop, true);
+  BOOST_CHECK_EQUAL(gridStart > gridStop, false);
+  BOOST_CHECK_EQUAL(gridStart >= gridStop, false);
+
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), nBins + 2ul);
+  auto itr = gridStart++;
+  BOOST_CHECK_EQUAL(std::distance(itr, gridStart), 1ul);
+  BOOST_CHECK_EQUAL(std::distance(itr, gridStop), nBins + 2ul);
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), nBins + 1ul);
+
+  itr = ++gridStart;
+  BOOST_CHECK_EQUAL(std::distance(itr, gridStart), 0ul);
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), nBins);
+
+  itr = gridStart + std::distance(gridStart, gridStop);
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), nBins);
+  BOOST_CHECK_EQUAL(std::distance(itr, gridStop), 0ul);
+  BOOST_CHECK_EQUAL(itr == gridStop, true);
+  
+  itr = gridStart - 1ul;
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), nBins);
+  BOOST_CHECK_EQUAL(std::distance(itr, gridStop), nBins + 1ul);
+
+  gridStart += std::distance(gridStart, gridStop);
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), 0ul);
+  BOOST_CHECK_EQUAL(gridStart == gridStop, true);
+  
+  gridStart -= 3ul;
+  BOOST_CHECK_EQUAL(std::distance(gridStart, gridStop), 3ul);
+  BOOST_CHECK_EQUAL(gridStart != gridStop, true);
+
+  [[maybe_unused]] double value = *gridStart;
```

Comment:
Isn't this, in  fact, always unused?

Reply:
yes indeed. This is for testing the `operator*` doesn't have sudden crushes. We do nothing with `value`

As far as I know c++ has this `maybe_unused` attribute. Is there as well a `unused` attribute, too? If so, I'm not aware of it